package.json

```diff
@@ -4,9 +4,9 @@
     "description": "home of the osu! wiki",
     "repository": "github:ppy/osu-wiki",
     "dependencies": {
-        "remark": "12.0.0",
-        "remark-cli": "8.0.0",
-        "remark-frontmatter": "2.0.0",
+        "remark": "^13.0.0",
+        "remark-cli": "^9.0.0",
+        "remark-frontmatter": "^3.0.0",
```

Comment:
I personally wouldn't go as far as to add carets here. Consider the case of a contributor that does `npm install` to run remark locally, but doesn't really understand much about what they're doing, and then accidentally bumps and commits the lockfile to a PR, because one of these packages has a new minor/bugfix release.

Back when I hard-pinned these the explicit intention was to use specific known-good versions and combine that with npm audits to check security advisories about them, to preserve as much reproducibility in the setup as is possible.

Reply:
true. besides, remarkjs/remark-lint@7791db9450da83e402455c902b6b202d178133ba looks like I need to update a couple more packages

Reply:
maybe I'm misunderstanding something but this is exactly what `package-lock` is for-- `package` specifies what's compatible with your project, `package-lock` pins it to something so that every contributor's `npm install` result is the same. caret is the default and standard for almost everything on npm bc no breaking changes are made within major versions, I don't see why it wouldn't go here too

Reply:
I thought that running `npm install` *will rewrite* `package-lock.json` if it determines that it can (and from limited experience, I did witness it doing that). Googling around gives mixed results - some claim that the above behaviour is fixed - but I still feel hard pinning is safest. Especially that the circumstances here are special: remark is a tool here, and not a library dependency.